src/Stratis.Bitcoin.Features.Wallet/WalletTransactionHandler.cs

```diff
@@ -208,9 +213,21 @@ private void AddSecrets(TransactionBuildContext context)
                 return;
 
             Wallet wallet = this.walletManager.GetWalletByName(context.AccountReference.WalletName);
-
+            Key privateKey;
             // get extended private key
-            var privateKey = Key.Parse(wallet.EncryptedSeed, context.WalletPassword, wallet.Network);
+            string cacheKey = $"{wallet.EncryptedSeed}{context.WalletPassword}{wallet.Network}";
```

Comment:
`context.WalletPassword` should be a secure string on its own.
And not a good idea to use the password as cash key that will make it vulnerable as the password can easily extract the key.

Reply:
Well, I took it from your conversation on issue #1214 , If I won't use it as combined key, than anyone can fake cacheKey and get Private Key from cache, or not? @dangershony

Reply:
I think retrieving from cache is not that hard anyway even without the password, the point of secure string is that the content is encrypted in memory (and if cache is persisted to disk) so by keeping the password in clear memory we expose the key (as also the encrypted seed is in clear memory.

We need to keep the password as SecureString as well, the only pint where the password/private seed can be not as SecureString is at the point of deriving the prv key and the point we read it form the controller.